magento/magento2
# GraphQl Product visibility for search results

### Description (*)

Product Visibility handling for search feature ,

### Related Pull Requests

Fixes for https://github.com/magento/magento2/issues/36591

### Fixed Issues (if relevant)

1. Fixes magento/magento2#36591

### Manual testing scenarios (*)

1. SKU `API-TEST-1`, has assigned a visibility as Catalog, so it will visible only with checking category condition.
![Admin](https://user-images.githubusercontent.com/104893739/208025694-fceccec9-29a9-4570-bdad-496fdbbb3772.png)
2. Result won't be appear if we search this sku.
![Search_without_filter](https://user-images.githubusercontent.com/104893739/208025730-eae8126c-2a4c-46a9-a010-bc70b9e7e1d7.png)
3. Result will appear by checking the combination of assigned category and search key word
![Search_with_filter](https://user-images.githubusercontent.com/104893739/208025751-51da8826-9147-4315-a69f-3cf1cbbf4210.png)


### Questions or comments


### Contribution checklist (*)
